### PR TITLE
HPCC-23644 Containerized component logging logic

### PR DESCRIPTION
**Pull request by rpastrana**
- Provides simple setupContainerizedLoggingMsgHandler function
- Conditionally removes creation of component log file logic
- Logging to continue on default stderr stream log handler
- Per container customizations are applied to stderr log handler
- Clients and test code continues to log to file
- Provides per component logging configuration support
- Adds log class and log audience string helper methods

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
This pull request replaces https://github.com/hpcc-systems/HPCC-Platform/pull/13500

**Comment by rpastrana** [comment]
@jakesmith please review

**Comment by jakesmith** [comment]
@rpastrana - a few minor comments, could be merged now I think, but will test issue I saw previously has cleared now rebased.

**Comment by jakesmith** [comment]
@rpastrana - looks good, will need a change to values.schema.json to allow logging: properties to be used, but can be separate PR.

**Comment by rpastrana** [comment]
@jakesmith agreed: https://track.hpccsystems.com/browse/HPCC-23824

**Comment by rpastrana** [comment]
@richardkchapman please merge

File: roxie/ccd/ccdmain.cpp [hunk 1]
```diff
@@ -543,6 +543,8 @@ version: "1.0"
     - name: workunit
       port: 0
       numThreads: 0
+  logging:
+      detail: 100
```
Comment (richardkchapman) [inline roxie/ccd/ccdmain.cpp:547]: What does this correspond to, and how does it related to the existing traceLevel setting

File: system/jlib/jlog.cpp [hunk 2]
```diff
@@ -2324,6 +2324,75 @@ MODULE_EXIT()
     thePassAllFilter = NULL;
 }
 
+#ifdef _CONTAINERIZED
+
+static constexpr const char * logFielsdAtt = "@fields";
```
Comment (richardkchapman) [inline system/jlib/jlog.cpp:2329]: Typo fields

**Pull request merged**